nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,158 @@
+            .name("delete-hb-separator")
+            .displayName("Delete Row Key Separator")
+            .description("The separator character(s) that separate multiple row keys " +
+                    "when multiple row keys are provided in the flowfile body")
+            .required(true)
+            .defaultValue(",")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(false)
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = super.getSupportedPropertyDescriptors();
+        properties.add(ROW_ID_LOCATION);
+        properties.add(FLOWFILE_FETCH_COUNT);
+        properties.add(BATCH_SIZE);
+        properties.add(KEY_SEPARATOR);
+
+        return properties;
+    }
+
+    @Override
+    protected void doDelete(ProcessContext context, ProcessSession session) throws Exception {
+        final int batchSize      = context.getProperty(BATCH_SIZE).asInteger();
+        final String location    = context.getProperty(ROW_ID_LOCATION).getValue();
+        final int flowFileCount  = context.getProperty(FLOWFILE_FETCH_COUNT).asInteger();
+        List<FlowFile> flowFiles = session.get(flowFileCount);
+
+        if (flowFiles != null && flowFiles.size() > 0) {
+            for (int index = 0; index < flowFiles.size(); index++) {
+                FlowFile flowFile = flowFiles.get(index);
+                try {
+                    if (location.equals(ROW_ID_BODY.getValue())) {
+                        doDeleteFromBody(flowFile, context, session, batchSize);
+                    } else {
+                        doDeleteFromAttribute(flowFile, context);
+                    }
+                    session.transfer(flowFile, REL_SUCCESS);
+                } catch (Exception ex) {
+                    getLogger().error(ex.getMessage(), ex);
+                    session.transfer(flowFile, REL_FAILURE);
+                }
+            }
+        }
+    }
+
+    private void doDeleteFromAttribute(FlowFile flowFile, ProcessContext context) throws Exception {
+        String rowKey = context.getProperty(ROW_ID).evaluateAttributeExpressions(flowFile).getValue();
+        String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        clientService.delete(tableName, rowKey.getBytes("UTF-8"));
+    }
+
+    private void doDeleteFromBody(FlowFile flowFile, ProcessContext context, ProcessSession session, int batchSize) throws Exception {
+        String keySeparator = context.getProperty(KEY_SEPARATOR).getValue();
+        String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+
+        List<byte[]> deletes = new ArrayList<>();
+        session.read(flowFile, in -> {
+            String data = IOUtils.toString(in, "UTF-8");
+            String[] lines = data.split("[\\r\\n]");
```

Comment:
we already have the separator, do we need this? Moreover if the line terminator is `\r\n` this would generate an empty key to be deleted too IIUC. Since HBase keys are bytes `\r` and `\n` may be valid values inside a key the user wants to delete... I'd avoid doing this.